pkg/apis/org/v1/backuprestorecommon_types.go

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package v1
+
+const (
+	USERNAME_SECRET_KEY              = "username"
+	PASSWORD_SECRET_KEY              = "password"
+	RESTIC_REPO_PASSWORD_SECRET_KEY  = "repo-password"
+	SSH_PRIVATE_KEY_SECRET_KEY       = "ssh-privatekey"
+	AWS_ACCESS_KEY_ID_SECRET_KEY     = "awsAccessKeyId"
+	AWS_SECRET_ACCESS_KEY_SECRET_KEY = "awsSecretAccessKey"
+
+	STATE_IN_PROGRESS = "InProgress"
+	STATE_SUCCEEDED   = "Succeeded"
+	STATE_FAILED      = "Failed"
+)
+
+// +k8s:openapi-gen=true
+// List of supported backup servers
+type BackupServersConfigs struct {
+	// Sftp backup server configuration.
+	// Mandatory fields are: Username, Hostname, RepositoryPath, RepositoryPasswordSecretRef, SshKeySecretRef.
+	// +optional
+	Sftp *SftpServerConfing `json:"sftp,omitempty"`
+	// Rest backup server configuration.
+	// Mandatory fields are: Hostname, RepositoryPasswordSecretRef.
+	// +optional
+	Rest *RestServerConfig `json:"rest,omitempty"`
+	// Amazon S3 or compatible alternatives.
+	// Mandatory fields are: RepositoryPasswordSecretRef, RepositoryPath, CredentialsSecretRef.
+	// +optional
+	AwsS3 *AwsS3ServerConfig `json:"awss3,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// SFTP backup server configuration
+// Example: user@host://srv/repo
+// Mandatory fields are: Username, Hostname, RepositoryPath, RepositoryPasswordSecretRef, SshKeySecretRef
```

Comment:
If field is mandatory then neither  `+optional` nor `omitempty`

Reply:
They are mandatory for the server configuration, but they are not mandatory for the CR

Reply:
Thanks, I see.